### PR TITLE
Remove unused variables

### PR DESCRIPTION
**Pull request by ignacioola**
Fix unused variable ESLint warnings.

Test Plan:
----------
Run:
```
yarn lint
```
Check the lint warning count is reduced.

Release Notes:
--------------
[INTERNAL] [ENHANCEMENT] [eslint] - Fix unused variable eslint warnings

### CODE REVIEW

File: local-cli/link/ios/registerNativeModule.js [hunk 1]
```diff
@@ -10,19 +10,16 @@
 const xcode = require('xcode');
 const fs = require('fs');
 const path = require('path');
-const log = require('npmlog');
 
 const addToHeaderSearchPaths = require('./addToHeaderSearchPaths');
 const getHeadersInFolder = require('./getHeadersInFolder');
 const getHeaderSearchPath = require('./getHeaderSearchPath');
-const getProducts = require('./getProducts');
```
Comment (TheSavior) [inline local-cli/link/ios/registerNativeModule.js]: Looks like `getProducts.js` isn't accessed anywhere else and can be deleted too.

**Comment by TheSavior** [comment]
cc @grabbou since this touches the CLI

File: local-cli/link/ios/registerNativeModule.js (hunk 1 above)
Comment (ignacioola) [reply local-cli/link/ios/registerNativeModule.js]: I'll remove it.

**Comment by grabbou** [comment]
Thanks for submitting this PR and @TheSavior for CCing me. In fact, we have discovered a bit more unused variables in CLI later down the road.

**Pull request closed without merging**